src/services/io/podio/CMakeLists.txt

```diff
@@ -31,17 +27,19 @@ plugin_glob_all(${PLUGIN_NAME})
 
 # Generate the datamodel header files.
 add_custom_command(
-        OUTPUT ${podio_BINARY_DIR}/datamodel_glue.h ${podio_BINARY_DIR}/datamodel_includes.h
-        COMMAND python3 ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py WORKING_DIR=${podio_BINARY_DIR} EDM4HEP_INCLUDE_DIR=${EDM4HEP_INCLUDE_DIR} EDM4EIC_INCLUDE_DIR=${EDM4EIC_INCLUDE_DIR}
+        OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/datamodel_glue.h ${CMAKE_CURRENT_BINARY_DIR}/datamodel_includes.h
+        COMMAND python3 ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
+                WORKING_DIR=${CMAKE_CURRENT_BINARY_DIR}
+                EDM4HEP_INCLUDE_DIR=$<REMOVE_DUPLICATES:$<TARGET_PROPERTY:EDM4HEP::edm4hep,INTERFACE_INCLUDE_DIRECTORIES>>
+                EDM4EIC_INCLUDE_DIR=$<REMOVE_DUPLICATES:$<TARGET_PROPERTY:EDM4EIC::edm4eic,INTERFACE_INCLUDE_DIRECTORIES>>
         MAIN_DEPENDENCY ${PROJECT_SOURCE_DIR}/make_datamodel_glue.py
 )
 
 # Add include directories (works same as target_include_directories)
-message(STATUS "fmt_INCLUDE_DIR=${fmt_INCLUDE_DIR}")
-plugin_include_directories(${PLUGIN_NAME} SYSTEM PUBLIC ${INCLUDE_DIRS} ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_BINARY_DIR}/src ${CMAKE_CURRENT_SOURCE_DIR} ${EICRECON_SOURCE_DIR}/src)
+plugin_include_directories(${PLUGIN_NAME} PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
Not sure that this is needed. We are not supposed to link from plugin to plugin.
```suggestion
```

Reply:
Confirmed that this is needed. Without this, `datamodel_glue.h` is not found by `src/services/io/podio/JEventProcessorPODIO.cc`.

Reply:
Reverted in https://github.com/eic/EICrecon/pull/720/commits/acb8e746f5b4744e9bc4871ba7eaa1e756835eb3

Reply:
```suggestion
plugin_include_directories(${PLUGIN_NAME} PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
```